internal/exec/engine.go

```diff
@@ -341,7 +339,9 @@ func (e *Engine) fetchProviderConfig() (types.Config, error) {
 // provided config will be returned unmodified. An updated fetcher will be
 // returned with any new timeouts set.
 func (e *Engine) renderConfig(cfg types.Config) (types.Config, error) {
-	if cfgRef := cfg.Ignition.Config.Replace; cfgRef.Source != nil {
+	cfgRef := cfg.Ignition.Config.Replace
+	sources := cfgRef.GetSources()
+	if len(sources) != 0 {
```

Comment:
nit: let's drop the `sources` variable assignment and in-line it in the `if` as it isn't re-used.